src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptor.java

```diff
@@ -0,0 +1,441 @@
+package org.broadinstitute.hellbender.cmdline.GATKPlugin;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.barclay.argparser.ArgumentCollection;
+import org.broadinstitute.barclay.argparser.CommandLineException;
+import org.broadinstitute.barclay.argparser.CommandLinePluginDescriptor;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.walkers.annotator.Annotation;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.reflections.ReflectionUtils;
+
+import java.lang.reflect.Modifier;
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A plugin descriptor for managing the dynamic discovery of both @{InfoFieldAnnotation} and @{GenotypeAnnotation} objects
+ * within the packages defined the method getPackageNames() (default {@link org.broadinstitute.hellbender.tools.walkers.annotator}).
+ * Also handles integrating annotation specific arguments from the command line with tool specified defaults.
+ *
+ * Unlike {@link GATKReadFilterPluginDescriptor} annotation order is not important and thus argument order is not guaranteed to be
+ * preserved in all cases, especially when group annotations are involved.
+ *
+ * An alternative method for discovering annotations would be to use ClassUtils.knownSubInterfaceSimpleNames(Annotation.class);
+ * Which is the preferred way to discover Annotations for testing purposes.
+ *
+ * NOTE: this class enforces that annotations with required arguments must see their arguments, yet this is not currently tested
+ *       as no such annotations exist in the GATK.
+ */
+public class GATKAnnotationPluginDescriptor  extends CommandLinePluginDescriptor<Annotation> {
+    //TODO this should be a configurable option or otherwise exposed to the user when configurations are fully supported.
+    private static final String pluginPackageName = "org.broadinstitute.hellbender.tools.walkers.annotator";
```

Comment:
Now that the config file is in, this can use the config file...

Reply:
This is true, I would say that we do this in a future PR once this is in. This intended to be the implementation of the basic functionality before it's hooked up to the rest of the codebase.

Reply:
Ok, but "basic" can also means that it is usable out of the GATK codebase, that is, downstream projects. For example, even if it is not integrated in the tools for annotation, an API user might want to use the plugin descriptor for its own annotation tool...

Reply:
But in my case, I can wait